src/qunit.cpp

```diff
@@ -123,16 +125,14 @@ void QUnit::Detach(bitLenInt start, bitLenInt length, QInterfacePtr dest)
         unit->Dispose(mapped, length);
     }
 
-    shards.erase(shards.begin() + start, shards.begin() + start + length);
+        shards.erase(shards.begin() + start, shards.begin() + start + length);
 
-    if (unit->GetQubitCount() == length) {
-        return;
-    }
-
-    /* Find the rest of the qubits. */
-    for (auto shard : shards) {
-        if (shard.unit == unit && shard.mapped > (mapped + length)) {
-            shard.mapped -= length;
+    if (unit->GetQubitCount() != length) {
+        /* Find the rest of the qubits. */
+        for (auto &&shard : shards) {
+            if (shard.unit == unit && shard.mapped > (mapped + length)) {
+                shard.mapped -= length;
+            }
```

Comment:
This all looks exactly the same as the old code; might as well leave it as it was.